File: saizad-mvvm/src/main/java/com/saizad/mvvm/ui/calendar/TimePickerFragment.java
package com.saizad.mvvm.ui.calendar;

import android.app.Dialog;
import android.app.TimePickerDialog;
import android.os.Bundle;
import android.widget.TimePicker;

import androidx.annotation.NonNull;

import org.jetbrains.annotations.NotNull;
import org.joda.time.DateTime;

public class TimePickerFragment extends BaseCalendarPickerFragment
                            implements TimePickerDialog.OnTimeSetListener {

    @NotNull
    private final DateTime dateTime;

    public TimePickerFragment(@NonNull DateTime dateTime, @NonNull DateSelectedListener dateSelectedListener) {
        super(dateSelectedListener);
        this.dateTime = dateTime;
    }

    @NotNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        return new TimePickerDialog(getActivity(), this, dateTime.getHourOfDay(), dateTime.minuteOfHour().get(),
                false);
    }

    public void onTimeSet(TimePicker view, int hourOfDay, int minute) {
        final DateTime dateTime = this.dateTime.withTimeAtStartOfDay().plusHours(hourOfDay).plusMinutes(minute);
        dateSelectedListener.selected(dateTime);
    }
}

File: saizad-mvvm/src/main/java/com/saizad/mvvm/ui/calendar/TimePickerFragment.kt
package com.saizad.mvvm.ui.calendar

import android.app.Dialog
import android.app.TimePickerDialog
import android.app.TimePickerDialog.OnTimeSetListener
import android.os.Bundle
import android.widget.TimePicker
import org.joda.time.DateTime

class TimePickerFragment(
    private val dateTime: DateTime,
    dateSelectedListener: (DateTime) -> Unit
) : BaseCalendarPickerFragment(dateSelectedListener), OnTimeSetListener {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return TimePickerDialog(
            activity, this, dateTime.hourOfDay, dateTime.minuteOfHour().get(),
            false
        )
    }

    override fun onTimeSet(view: TimePicker, hourOfDay: Int, minute: Int) {
        val dateTime =
            dateTime.withTimeAtStartOfDay().plusHours(hourOfDay).plusMinutes(minute)
        dateSelectedListener.invoke(dateTime)
    }

}